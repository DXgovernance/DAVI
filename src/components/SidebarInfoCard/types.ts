import { BigNumber } from 'ethers';

export interface SidebarInfoCardProps {
  proposalTime: BigNumber;
  quorum: number;
}
