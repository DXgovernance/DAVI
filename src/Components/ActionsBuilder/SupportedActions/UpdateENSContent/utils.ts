import contenthash from 'content-hash';
import { MAINNET_ID, LOCALHOST_ID, GOERLI_ID } from 'utils';
import { utils } from 'ethers';

const DEFAULT_NAMEHASH =
  '0x0000000000000000000000000000000000000000000000000000000000000000';

export const convertToNameHash = (name: string) => {
  if (!name) {
    return DEFAULT_NAMEHASH;
  }
  return utils.namehash(name);
};

export const convertToContentHash = (ipfsHash: string) => {
  if (!ipfsHash) {
    return '';
  }
  return contenthash.fromIpfs(ipfsHash);
};

export const convertToIpfsHash = (contentHash: string) => {
  if (!contentHash) return null;
  return contenthash.decode(contentHash);
};

export const getIpfsUrl = (ipfsHash: string) => {
  const ipfsRoot = 'ipfs://';
  return `${ipfsRoot}${ipfsHash}`;
};

export const isValidChainId = (chainId: number) => {
  if (!chainId || chainId === LOCALHOST_ID) {
    return MAINNET_ID;
  } else {
    return chainId;
  }
};

export const isAvailableOnENS = (chainId: number) => {
  const validChainId = isValidChainId(chainId);
  const ensNetworks = [MAINNET_ID, GOERLI_ID];
  return ensNetworks.includes(validChainId);
};
