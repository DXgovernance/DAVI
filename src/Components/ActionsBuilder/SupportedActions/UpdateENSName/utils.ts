import namehash from 'eth-ens-namehash';
import contenthash from 'content-hash';
import { useState, useEffect } from 'react';

export const convertToNameHash = (name: string) => namehash.hash(name);

export const convertToContentHash = (ipfsHash: string) =>
  contenthash.fromIpfs(ipfsHash);

export const convertToIPFSHash = (contentHash: string) => {
  if (!contentHash) return null;
  return contenthash.decode(contentHash);
};

export const useDebounce = (value: string, delay: number) => {
  const [debouncedValue, setDebouncedValue] = useState(value);

  useEffect(() => {
    const handler = setTimeout(() => {
      setDebouncedValue(value);
    }, delay);

    return () => {
      clearTimeout(handler);
    };
  }, [value, delay]);

  return debouncedValue;
};
