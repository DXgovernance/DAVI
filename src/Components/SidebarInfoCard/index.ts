export { default as SidebarInfoCard } from './SidebarInfoCard';
