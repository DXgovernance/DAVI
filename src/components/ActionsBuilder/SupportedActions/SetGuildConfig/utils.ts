import { FIELDS } from './constants';
import { BigNumber } from 'ethers';
import { SetGuildConfigFields, ControlField } from './types';

export const bn = (n: string | number | BigNumber) => BigNumber.from(n);

type UpdatedValuesReturn = {
  [key in ControlField]: {
    newValue: BigNumber;
    currentValue: BigNumber;
  };
};
export const getUpdatedValues = (
  current: SetGuildConfigFields,
  modifyed: SetGuildConfigFields
): UpdatedValuesReturn | {} => {
  if (!current || !modifyed) return {};
  return Object.keys(current).reduce((acc, key) => {
    if (!FIELDS.map(f => f.name).includes(key)) return acc;
    const currentValue = bn(current[key] || 0);
    const newValue = bn(modifyed[key] || 0);

    if (!currentValue.eq(newValue)) {
      return {
        ...acc,
        [key]: { newValue, currentValue },
      };
    }
    return acc;
  }, {});
};

export const pickUpdatedOrDefaultValue = (
  current: BigNumber,
  modifyed: BigNumber
): BigNumber => {
  if (BigNumber.isBigNumber(current)) {
    if (BigNumber.isBigNumber(modifyed) && !bn(modifyed).eq(current)) {
      return bn(modifyed);
    }
    return bn(current);
  } else if (BigNumber.isBigNumber(modifyed)) {
    return bn(modifyed);
  }
  return bn(0);
};
