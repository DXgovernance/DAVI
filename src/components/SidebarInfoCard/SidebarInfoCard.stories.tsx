import { ComponentStory, ComponentMeta } from '@storybook/react';
import { SidebarInfoCard } from 'components/SidebarInfoCard';
import { SidebarInfoCardProps } from './types';
import { BigNumber } from 'ethers';

export default {
  title: 'SidebarInfoCard',
  component: SidebarInfoCard,
  argTypes: {
    proposalTime: {
      description: 'Duration of the proposal',
    },
    quorum: {
      description: 'Quorum percentage',
    },
  },
} as ComponentMeta<typeof SidebarInfoCard>;

const sidebarInfoCardArgs: SidebarInfoCardProps = {
  proposalTime: BigNumber.from(15888),
  quorum: 1,
};

const Template: ComponentStory<typeof SidebarInfoCard> = args => (
  <SidebarInfoCard {...args} />
);

export const Simple = Template.bind({});
Simple.args = sidebarInfoCardArgs;

export const Loading = Template.bind({});
Loading.args = {};
