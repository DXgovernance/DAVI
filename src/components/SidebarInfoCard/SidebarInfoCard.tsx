import { duration } from 'moment';
import SidebarCard, {
  SidebarCardContent,
  SidebarCardHeader,
} from 'components/SidebarCard';
import { Loading } from 'components/Primitives/Loading';
import { Row, Label, ColoredLabel } from './SidebarInfoCard.styled';
import { SidebarInfoCardProps } from './types';
import { useTranslation } from 'react-i18next';

const SidebarInfoCard: React.FC<SidebarInfoCardProps> = ({
  proposalTime,
  quorum,
}) => {
  const { t } = useTranslation();

  return (
    <SidebarCard
      header={<SidebarCardHeader>{t('information')}</SidebarCardHeader>}
    >
      <SidebarCardContent>
        <Row>
          <Label>{t('consensusSystem')}</Label>
          <ColoredLabel>{t('guilds.guild')}</ColoredLabel>
        </Row>
        <Row>
          <Label>{t('proposalDuration')}</Label>
          <ColoredLabel>
            {proposalTime ? (
              duration(proposalTime?.toNumber(), 'seconds').humanize()
            ) : (
              <Loading style={{ margin: 0 }} loading text />
            )}
          </ColoredLabel>
        </Row>
        <Row>
          <Label>{t('quorum')}</Label>
          <ColoredLabel>
            {quorum != null ? (
              `${quorum}%`
            ) : (
              <Loading style={{ margin: 0 }} loading text />
            )}
          </ColoredLabel>
        </Row>
      </SidebarCardContent>
    </SidebarCard>
  );
};

export default SidebarInfoCard;
