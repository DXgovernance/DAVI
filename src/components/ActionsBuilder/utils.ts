import { DecodedAction } from './types';
import { Permission } from './types';
import Web3 from 'web3';
import { NULL_SIGNATURE } from 'utils';
import { SupportedAction } from './types';

const web3 = new Web3();

export const getPermissionArgs = (
  decodedActions: DecodedAction[]
): Permission[] => {
  const permissions = decodedActions.map(decodedAction => {
    // Native transfer of ETH does not have a function
    if (
      decodedAction?.decodedCall?.callType ===
        SupportedAction.NATIVE_TRANSFER ||
      decodedAction?.decodedCall?.callType === SupportedAction.RAW_TRANSACTION
    ) {
      return {
        from: decodedAction?.decodedCall?.from,
        to: decodedAction?.decodedCall?.to,
        callType: decodedAction?.decodedCall?.callType,
        functionSignature: NULL_SIGNATURE,
      };
    }
    const functionInputs = decodedAction?.decodedCall?.function?.inputs?.map(
      input => input.type
    );
    const functionName = decodedAction?.decodedCall?.function?.name;
    const functionSignature = web3.eth.abi.encodeFunctionSignature(
      `${functionName}(${functionInputs.join(',')})`
    );
    return {
      from: decodedAction?.decodedCall?.from,
      to: decodedAction?.decodedCall?.to,
      callType: decodedAction?.decodedCall?.callType,
      functionSignature,
    };
  });
  return permissions;
};
