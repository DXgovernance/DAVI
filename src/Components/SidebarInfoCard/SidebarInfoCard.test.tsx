import SidebarInfoCard from './SidebarInfoCard';
import { render } from 'utils/tests';
import { BigNumber } from 'ethers';

jest.mock('hooks/Guilds/ether-swr/guild/useGuildConfig', () => ({
  useGuildConfig: () => jest.fn(),
}));
jest.mock('contexts/index', () => jest.fn());

describe('SidebarInfoCard', () => {
  const mockProposalTime = BigNumber.from(2);
  it('matches information card snapshot', () => {
    const container = render(
      <SidebarInfoCard quorum={1} proposalTime={mockProposalTime} />
    );
    expect(container).toMatchSnapshot();
  });
});
