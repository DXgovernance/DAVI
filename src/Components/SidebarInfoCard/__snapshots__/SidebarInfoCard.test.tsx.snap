// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarInfoCard matches information card snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
       
      <div
        class="sc-bdfBQB sc-eCstlR bmhbKs chDmvy"
      >
        <div
          class="sc-bdfBQB sc-jSgvzq bmhbKs iPIXsc"
        >
          <h2
            class="sc-iBPTik sc-fubCzh cBZYjx gMMEXo"
          >
            information
          </h2>
        </div>
        <div
          class="sc-bdfBQB bmhbKs"
        >
          <div
            class="sc-bdfBQB sc-jrAFXE bmhbKs ldFlnI"
          >
            <div
              class="sc-dlfnuX sc-crrszt ghjVNo gDaPjc"
            >
              <span
                class="sc-dQoVA dZpTnI"
              >
                consensusSystem
              </span>
              <span
                class="sc-dQoVA sc-bqyKOL dZpTnI gqFPfm"
              >
                guilds.guild
              </span>
            </div>
            <div
              class="sc-dlfnuX sc-crrszt ghjVNo gDaPjc"
            >
              <span
                class="sc-dQoVA dZpTnI"
              >
                proposalDuration
              </span>
              <span
                class="sc-dQoVA sc-bqyKOL dZpTnI gqFPfm"
              >
                a few seconds
              </span>
            </div>
            <div
              class="sc-dlfnuX sc-crrszt ghjVNo gDaPjc"
            >
              <span
                class="sc-dQoVA dZpTnI"
              >
                quorum
              </span>
              <span
                class="sc-dQoVA sc-bqyKOL dZpTnI gqFPfm"
              >
                1%
              </span>
            </div>
          </div>
        </div>
      </div>
       
    </div>
  </body>,
  "container": <div>
     
    <div
      class="sc-bdfBQB sc-eCstlR bmhbKs chDmvy"
    >
      <div
        class="sc-bdfBQB sc-jSgvzq bmhbKs iPIXsc"
      >
        <h2
          class="sc-iBPTik sc-fubCzh cBZYjx gMMEXo"
        >
          information
        </h2>
      </div>
      <div
        class="sc-bdfBQB bmhbKs"
      >
        <div
          class="sc-bdfBQB sc-jrAFXE bmhbKs ldFlnI"
        >
          <div
            class="sc-dlfnuX sc-crrszt ghjVNo gDaPjc"
          >
            <span
              class="sc-dQoVA dZpTnI"
            >
              consensusSystem
            </span>
            <span
              class="sc-dQoVA sc-bqyKOL dZpTnI gqFPfm"
            >
              guilds.guild
            </span>
          </div>
          <div
            class="sc-dlfnuX sc-crrszt ghjVNo gDaPjc"
          >
            <span
              class="sc-dQoVA dZpTnI"
            >
              proposalDuration
            </span>
            <span
              class="sc-dQoVA sc-bqyKOL dZpTnI gqFPfm"
            >
              a few seconds
            </span>
          </div>
          <div
            class="sc-dlfnuX sc-crrszt ghjVNo gDaPjc"
          >
            <span
              class="sc-dQoVA dZpTnI"
            >
              quorum
            </span>
            <span
              class="sc-dQoVA sc-bqyKOL dZpTnI gqFPfm"
            >
              1%
            </span>
          </div>
        </div>
      </div>
    </div>
     
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
